src/morphology.rs

```diff
@@ -597,12 +619,23 @@ impl Mask {
     ) -> impl Iterator<Item = &'a Luma<u8>> {
         self.elements
             .iter()
-            .map(move |(i, j)| (x as i64 + *i as i64, y as i64 + *j as i64))
+            .map(move |p| (x as i64 + p.x as i64, y as i64 + p.y as i64))
             .filter(move |(i, j)| {
                 0 <= *i && *i < image.width() as i64 && 0 <= *j && *j < image.height() as i64
             })
             .map(move |(i, j)| image.get_pixel(i as u32, j as u32))
     }
+
+    /// all the slices are garanteed to be non empty
+    fn lines(&self) -> impl Iterator<Item = (i16, Vec<i16>)> + '_ {
+        self.elements
+            .iter()
+            .group_by(|p| p.y)
+            .into_iter()
+            .map(|(y, g)| (y, g.map(|p| p.x).collect()))
+            .collect::<Vec<_>>()
```

Comment:
Why collect? Just return after `map`

Reply:
not sure i fully understand, but basically it's a lifetime issue with GroupBy. Groupby is an iterable, not an iterator, thus issues with ownership

![error_groupby](https://github.com/image-rs/imageproc/assets/104371839/011a2a70-69c0-4b56-81b7-3f81aed5bc1e)